test/ref_ops_test.cpp

```diff
@@ -3560,6 +3560,46 @@ TEST_CASE(reshape_test)
     }
 }
 
+TEST_CASE(reverse_test_axis1)
+{
+    migraphx::shape in_shape{migraphx::shape::float_type, {2, 16}};
+    std::vector<float> data(32);
+    std::iota(data.begin(), data.end(), 1);
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    auto l   = mm->add_literal(migraphx::literal{in_shape, data});
+    int axis = 1;
+    mm->add_instruction(migraphx::make_op("reverse", {{"axis", axis}}), l);
+    p.compile(migraphx::ref::target{});
+    auto result = p.eval({}).back();
+    std::vector<float> results_vector;
+    result.visit([&](auto output) { results_vector.assign(output.begin(), output.end()); });
+    std::vector<float> target_data = {16, 15, 14, 13, 12, 11, 10, 9,  8,  7,  6,
```

Comment:
It might be easier to make a copy of the original data, then call std::reverse on it. This way if the test changes in the future, it will be easier to manage.

Reply:
What kind of test changes do you refer to?

Reply:
right now, you have the target_data values hardcoded. I'm suggesting to generate these numbers and call std::reverse on them.

Reply:
What is the benefit? Hardcoded is less computation vs generating copy and reversing it is additional computation. I am not seeing clearly the benefit of it.

Reply:
It's preference, but if the test changes or becomes more complex, you would have to redo the values. The other option is to just reverse an existing vector in the expected behavior. I'm okay either way for this PR.

Reply:
Using `std::reverse` would help ensure the data is correct, whereas hardcoded data may not always be obvious.

Reply:
I understand but then with the same idea, we should not have had any hardcoded gold vectors in these tests, e.g. round_test or pow_test, there are many of them, for which we have standard functions to use.

Reply:
Replaced gold vectors with `std::reverse` and `std::swap_ranges` depending on axis. Retested, looks ok.

Reply:
> I understand but then with the same idea, we should not have had any hardcoded gold vectors in these tests, e.g. round_test or pow_test, there are many of them, for which we have standard functions to use.

Right, I think moving forward once we refactor other tests, we should replace the hardcoded values there as well.